File: 2022/2nd/solution.cpp
#include <fstream>
#include <iostream>

int get_result(int opp, int you)
{
    if (opp==you) return 1;
    else if (you>>2&opp|you<<1&opp) return 0;
    else if (you<<2&opp|you>>1&opp) return 2;
    else return -1;
}

void main()
{
    std::string row = "";
    std::ifstream file ("input.txt");
    int points = 0;
    while (getline(file, row)) 
    {
        int opp = 1;
        int you = 1;
        if (row.size() > 3)
        {
            opp <<= (int)row.at(0) - 65;
            you <<= (int)row.at(2) - 88;
            points += get_result(opp, you)*3 + (you>>1)+1;
        }
    }
    std::cout << points << std::endl;
}

File: 2022/2nd/solution.m
opp_shapes = ["A" "B" "C"].';
your_shapes = ["X" "Y" "Z"];
shape_points =  [1 2 3];
result = [1 2 0; 0 1 2; 2 0 1];
result_factor = 3;
result_points = result*result_factor;

strategy_guide = readmatrix("input.txt", "OutputType", "string");
points = 0;
points_b = 0;
for i = 1:size(strategy_guide, 1)
    opp = find(opp_shapes == strategy_guide(i, 1));
    you = find(your_shapes == strategy_guide(i, 2));
    points = points + shape_points(you) + result_points(opp, you);
    shape = find(result(opp,:) == you-1);
    points_b = points_b + result_points(2, you) + shape_points(x) ;
end
points
points_b
